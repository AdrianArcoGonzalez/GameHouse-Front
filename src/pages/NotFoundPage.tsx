import { useEffect } from "react";
import { NavLink } from "react-router-dom";
import heroTexts from "../components/HeroSection/heroData/heroData";
import HeroSection from "../components/HeroSection/HeroSection";
import NotFoundPageStyled from "./NotFoundPageStyled";

const NotFoundPage = (): JSX.Element => {
  useEffect(() => {
    window.scrollTo({ top: 0 });
  }, []);
  return (
    <>
      <HeroSection srcImage="/images/notFound.webp" text={heroTexts.notFound} />
      <NotFoundPageStyled>
        <span className="not-found__title">404 Page Not Found</span>

        <NavLink to={"/home"} className="not-found__link">
          Go Home
        </NavLink>
      </NotFoundPageStyled>
    </>
  );
};

export default NotFoundPage;
