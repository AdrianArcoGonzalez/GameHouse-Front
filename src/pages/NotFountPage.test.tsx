import { BrowserRouter } from "react-router-dom";

import TestRenderer from "react-test-renderer";
import NotFoundPage from "./NotFoundPage";

describe("Given a NotFoundPage component", () => {
  describe("When it's instantiated", () => {
    test("Then it should match the the NotFoundPage page screenshot", () => {
      const expectedNotFoundPage = TestRenderer.create(
        <BrowserRouter>
          <NotFoundPage />
        </BrowserRouter>
      );

      expect(expectedNotFoundPage).toMatchSnapshot();
    });
  });
});
