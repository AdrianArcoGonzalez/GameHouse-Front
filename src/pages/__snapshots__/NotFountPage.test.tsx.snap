// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a NotFoundPage component When it's instantiated Then it should match the the NotFoundPage page screenshot 1`] = `
Array [
  <section
    className="sc-bczRLJ dVNtAm"
    data-testid="section"
  >
    <img
      alt="backgroundImage"
      className="hero__image"
      height="200px"
      src="/images/notFound.webp"
      width="100vw"
    />
    <div
      className="hero-container"
    >
      <p
        className="hero-container__sentence"
      >
        “Who are you, that do not know your history?” (Fallout New Vegas)
      </p>
    </div>
  </section>,
  <div
    className="sc-gsnTZi dKokb"
  >
    <span
      className="not-found__title"
    >
      404 Page Not Found
    </span>
    <a
      className="not-found__link"
      href="/home"
      onClick={[Function]}
    >
      Go Home
    </a>
  </div>,
]
`;
